usb.c

```diff
@@ -1120,8 +1121,12 @@ int usb_context_scan(struct iio_scan_backend_context *ctx,
 		unsigned int interface = 0;
 
 		ret = libusb_open(dev, &hdl);
-		if (ret)
+		if (ret) {
+			if (ret != LIBUSB_ERROR_ACCESS)
+				perms = true;
 			continue;
+		}
+		perms = true;
```

Comment:
Yes - I'm not pleased with the state either, however without walking through sysfs (Linux only, and only for primary contexts), there isn't any way to see what is going on. 

My biggest issue - is just because you have one udev rule installed (by pid/vid for Pluto), means you will not get the warning - and still will not see a different PID/VID (like M2k).

I will revert the libiio changes - but other than printing a warning in libiio about not running as root/administrator - there isn't much you can do - you don't know if you are seeing the actual device you are looking for (M2k), or the device that the udev was installed for (Pluto). 

I guess - we should install an ATTRS{interface}=="IIO" udev rule, and just be done? Or is the pointer/printf about root worth it? It still might be - not sure.